src/search.F90

```diff
@@ -33,7 +33,8 @@ function binary_search_real(array, n, val) result(array_index)
     R = n
 
     if (val < array(L) .or. val > array(R)) then
-      call fatal_error("Value outside of array during binary search")
+      array_index = -1
```

Comment:
Is this an error code? Perhaps we want to define some constant error codes rather than return some arbitrary values for the array index?